src/gui/mzroll/controller.cpp

```diff
@@ -125,6 +138,69 @@ void Controller::updateMavenParameters(const QString& key,  const QVariant& valu
     if(value.type() == QVariant::Bool)
         data = std::to_string(value.toBool());
 
-    mw->mavenParameters->setPeakDetectionSettings(key.toLocal8Bit().data(),data.c_str());
-    mw->mavenParameters->setOptionsDialogSettings(key.toLocal8Bit().data(),data.c_str());
+    _mw->mavenParameters->setPeakDetectionSettings(key.toLocal8Bit().data(),data.c_str());
+    _mw->mavenParameters->setOptionsDialogSettings(key.toLocal8Bit().data(),data.c_str());
+}
+
+void Controller::_updateSettingsForSave(const QString& k, const QVariant& v)
+{
+    variant var;
+
+    if (QString(v.typeName()).contains("QDoubleSpinBox"))
+        var = v.value<QDoubleSpinBox*>()->value();
+
+    if (QString(v.typeName()).contains("QSpinBox"))
+        var = v.value<QSpinBox*>()->value();
+
+    if (QString(v.typeName()).contains("QSlider"))
+        var = v.value<QSlider*>()->value();
+
+    if (QString(v.typeName()).contains("QGroupBox"))
+        var = static_cast<int>(v.value<QGroupBox*>()->isChecked());
+
+    if (QString(v.typeName()).contains("QCheckBox"))
+        var = static_cast<int>(v.value<QCheckBox*>()->isChecked());
+
+    if (QString(v.typeName()).contains("QComboBox"))
+        var = v.value<QComboBox*>()->currentIndex();
+
+    if (QString(v.typeName()).contains("QTabWidget"))
+        var = v.value<QTabWidget*>()->currentIndex();
+
+    if (QString(v.typeName()).contains("QLineEdit"))
+        var = v.value<QLineEdit*>()->text().toStdString();
+
+    // specifically for massCutoffType
+    if (QString(v.typeName()).contains("QString"))
+        var = v.value<QString*>()->toStdString();
+
+    _mw->fileLoader->insertSettingForSave(k.toStdString(), var);
```

Comment:
`insertSettingForSave` can also be done within `_syncMpWithUi`, right? Just trying to understand.

Reply:
I think `_updateSettingsForSave` exists separately, even though it performs the same checks as `_syncMpWithUi`, because it is called every time the UI is explicitly changed (such as when startup settings are loaded or when `MavenParameters` is reset). `_syncMpWithUi` instead is a function that is only called to synchronize `MavenParameters` with the UI. This way they are semantically different. We can with some more modifications make the sync function perform insertions, but the call graph would get increasing ugly.